azurerm/internal/services/cosmos/resource_arm_cosmosdb_sql_database.go

```diff
@@ -79,12 +80,11 @@ func resourceArmCosmosDbSQLDatabaseCreate(d *schema.ResourceData, meta interface
 				return fmt.Errorf("Error checking for presence of creating Cosmos SQL Database %s (Account %s): %+v", name, account, err)
 			}
 		} else {
-			id, err := azure.CosmosGetIDFromResponse(existing.Response)
-			if err != nil {
+			if existing.ID != nil && *existing.ID != "" {
```

Comment:
we want to raise an error if there's already an existing resource (e.g. an ID is returned) so this wants to become:

```suggestion
			if existing.ID == nil || *existing.ID == "" {
```